plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1123,7 +1122,10 @@ private void executeExpireSnapshots(ConnectorSession session, IcebergTableExecut
                 IcebergSessionProperties.EXPIRE_SNAPSHOTS_MIN_RETENTION);
 
         long expireTimestampMillis = session.getStart().toEpochMilli() - retention.toMillis();
-        expireSnapshots(table, expireTimestampMillis, session, executeHandle.getSchemaTableName());
+
+        table.expireSnapshots()
```

Comment:
So actually I did it the same way originally but then I got this comment from @rdblue 
https://github.com/trinodb/trino/pull/10810#issuecomment-1044765060

Reply:
current implementation is based on iceberg's spark extention expire_snapshots